HopsanCore/src/Parameters.cpp

```diff
@@ -710,17 +710,22 @@ bool ParameterEvaluatorHandler::evaluateParameter(const HString &rName, HString
 {
     bool success = false;
     // Try our own parameters
-    for(size_t i = 0; i < mParameters.size(); ++i)
+    if(rName.startsWith("self#"))
```

Comment:
I am not sure if I like this, now "self./#" is required to self reference, this would feel odd when you write code and do something like pComponent->evaluateParameter("self#kp"), Now there are two "this"  pComponent and self#.

Also I am not sure if "self" belongs in here, the parameter evaluator should not need to know about "this", but I am not sure since it already know about "parent system". 

Also we should maybe try to use . instead of # for new things even in the core. I do not remember why I used # before.

Reply:
I thought this was what we agreed on; if you want to reference a parameter in the same component you should prepend "self." to avoid confusion. 

The . was replaced by # automatically somewhere. Perhaps it always replaces all dots with hashes before evaluating the parameters?

Reply:
I was thinking that self. should only be necessary if a name collission (ambiguity) is detected. Similarly to how the this pointer may be needed in a c++ class function. Say you have a member variable 'monkey' and a function parameter 'monkey'. Then you must place this-> infront of the member. But if the function parameter (argument) is called 'dog' then a this is not needed.
But maybe it is more cleare if the user must allways write self. explicitly. There is less risk for missinterpretation.

Reply:
The hash thing is fot backward compatibility. Maybe we can try tp solve this later. In a release that breaks this compatibility.